internal/internal_workflow.go

```diff
@@ -661,9 +667,27 @@ func executeDispatcher(ctx Context, dispatcher dispatcher, timeout time.Duration
 		return
 	}
 
-	us := getWorkflowEnvOptions(ctx).getUnhandledSignalNames()
+	weo := getWorkflowEnvOptions(ctx)
+	us := weo.getUnhandledSignalNames()
 	if len(us) > 0 {
-		env.GetLogger().Info("Workflow has unhandled signals", "SignalNames", us)
+		env.GetLogger().Warn("Workflow has unhandled signals", "SignalNames", us)
```

Comment:
Hrmm. We are warning in other SDKs on signals _with handlers_ whose handlers don't complete, but we are not warning on just any signal coming in that has no handler. Arguably to be more effective here we'd differentiate signals that users don't care about from ones they do, i.e. keeping track of all signal channels requested, and only warn on the latter. But I don't mind this change too, but it'll warn on signals the workflow doesn't care about unlike other SDKs. Maybe that's ok.

Reply:
I think it is hard to assume a users intent here.  Just  because they didn't request a signal channel doesn't mean they didn't  care about, they may have just have not accessed the channel by the time the workflow completed. We were already logging on a all signal channels so I left it as is.

Reply:
:+1: It is a bit different than other SDKs, but that's probably fine. I wonder if we should consider in other SDKs to _also_ warn on signals that the workflow has no handlers.